### PR TITLE
make unit tests work

### PR DESCRIPTION
**Pull request by davidharrigan**
I was not able to run unittests on master

Change here is to make `python setup.py test` be able to run unit tests.  I guess we could also do the python path trickery in `manage.py`.  I also had to move the packages referenced in the tests/ module up a level - `tests` => `tests.tests`. 

```
make test
cd tests && DJANGO_SETTINGS_MODULE=tests.sqlite_settings  ./manage.py test tests --verbosity=2
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/dharrigan/work/src/django-optimistic-lock/venv/lib/python3.4/site-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/Users/dharrigan/work/src/django-optimistic-lock/venv/lib/python3.4/site-packages/django/core/management/__init__.py", line 328, in execute
    django.setup()
  File "/Users/dharrigan/work/src/django-optimistic-lock/venv/lib/python3.4/site-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/dharrigan/work/src/django-optimistic-lock/venv/lib/python3.4/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models(all_models)
  File "/Users/dharrigan/work/src/django-optimistic-lock/venv/lib/python3.4/site-packages/django/apps/config.py", line 198, in import_models
    self.models_module = import_module(models_module_name)
  File "/Users/dharrigan/work/src/django-optimistic-lock/venv/lib/python3.4/importlib/__init__.py", line 109, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 2254, in _gcd_import
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1200, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1129, in _exec
  File "<frozen importlib._bootstrap>", line 1471, in exec_module
  File "<frozen importlib._bootstrap>", line 321, in _call_with_frames_removed
  File "/Users/dharrigan/private/django-optimistic-lock/tests/tests/models.py", line 3, in <module>
    from ool import VersionField, VersionedMixin
ImportError: No module named 'ool'
make: *** [test] Error 1
```

Feel free to punt and close this PR, or if you have suggestions, I'd be more than happy to rework this.

Thanks!

**Test run**
```
make test
DJANGO_SETTINGS_MODULE=tests.tests.sqlite_settings python setup.py test
running test
Searching for django<1.9,>=1.6b1
Reading https://pypi.python.org/simple/django/
Best match: Django 1.8.17
Downloading https://pypi.python.org/packages/44/41/bf93934082e9897a56a591a67bacbd9fb74e71244f3f42253432a9e627e6/Django-1.8.17.tar.gz#md5=e76842cdfbcb31286bd44f51e087a04c
Processing Django-1.8.17.tar.gz
Writing /var/folders/gn/wpplgzmx0z7ct_bjwh122q9c0000gn/T/easy_install-6rz_bnyx/Django-1.8.17/setup.cfg
Running Django-1.8.17/setup.py -q bdist_egg --dist-dir /var/folders/gn/wpplgzmx0z7ct_bjwh122q9c0000gn/T/easy_install-6rz_bnyx/Django-1.8.17/egg-dist-tmp-huajktk2
warning: no previously-included files matching '__pycache__' found under directory '*'
warning: no previously-included files matching '*.py[co]' found under directory '*'
creating /Users/dharrigan/private/django-optimistic-lock/.eggs/Django-1.8.17-py3.4.egg
Extracting Django-1.8.17-py3.4.egg to /Users/dharrigan/private/django-optimistic-lock/.eggs

Installed /Users/dharrigan/private/django-optimistic-lock/.eggs/Django-1.8.17-py3.4.egg
Searching for coverage
Reading https://pypi.python.org/simple/coverage/
Best match: coverage 4.2
Downloading https://pypi.python.org/packages/f5/47/2e39972c41eb988d551bb2da25c4ee16ea78492ec1b6ffdf09f50cd09b92/coverage-4.2.zip#md5=9841c9e1ddc862144fda3fd7e5ba5106
Processing coverage-4.2.zip
Writing /var/folders/gn/wpplgzmx0z7ct_bjwh122q9c0000gn/T/easy_install-7qy_jot_/coverage-4.2/setup.cfg
Running coverage-4.2/setup.py -q bdist_egg --dist-dir /var/folders/gn/wpplgzmx0z7ct_bjwh122q9c0000gn/T/easy_install-7qy_jot_/coverage-4.2/egg-dist-tmp-qxjv5z_s
warning: no previously-included files found matching 'ci/appveyor.token'
no previously-included directories found matching 'doc/_build'
no previously-included directories found matching 'tests/eggsrc/build'
no previously-included directories found matching 'tests/eggsrc/dist'
no previously-included directories found matching 'tests/eggsrc/*.egg-info'
warning: no previously-included files matching '*.py[co]' found anywhere in distribution
creating /Users/dharrigan/private/django-optimistic-lock/.eggs/coverage-4.2-py3.4-macosx-10.6-intel.egg
Extracting coverage-4.2-py3.4-macosx-10.6-intel.egg to /Users/dharrigan/private/django-optimistic-lock/.eggs

Installed /Users/dharrigan/private/django-optimistic-lock/.eggs/coverage-4.2-py3.4-macosx-10.6-intel.egg
running egg_info
creating django_optimistic_lock.egg-info
writing django_optimistic_lock.egg-info/PKG-INFO
writing requirements to django_optimistic_lock.egg-info/requires.txt
writing top-level names to django_optimistic_lock.egg-info/top_level.txt
writing dependency_links to django_optimistic_lock.egg-info/dependency_links.txt
writing manifest file 'django_optimistic_lock.egg-info/SOURCES.txt'
reading manifest file 'django_optimistic_lock.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.html' under directory 'tests/templates/'
writing manifest file 'django_optimistic_lock.egg-info/SOURCES.txt'
running build_ext
Creating test database for alias 'default'...
Creating test database for alias 'default'...
................s
----------------------------------------------------------------------
Ran 17 tests in 0.351s

OK (skipped=1)
Destroying test database for alias 'default'...
Destroying test database for alias 'default'...
```

```
make coverage                                                                                      1 ↵
DJANGO_SETTINGS_MODULE=tests.tests.sqlite_settings coverage run --source="./ool" setup.py test
running test
running egg_info
writing django_optimistic_lock.egg-info/PKG-INFO
writing top-level names to django_optimistic_lock.egg-info/top_level.txt
writing dependency_links to django_optimistic_lock.egg-info/dependency_links.txt
writing requirements to django_optimistic_lock.egg-info/requires.txt
reading manifest file 'django_optimistic_lock.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.html' under directory 'tests/templates/'
writing manifest file 'django_optimistic_lock.egg-info/SOURCES.txt'
running build_ext
Creating test database for alias 'default'...
Creating test database for alias 'default'...
................s
----------------------------------------------------------------------
Ran 17 tests in 0.450s

OK (skipped=1)
Destroying test database for alias 'default'...
Destroying test database for alias 'default'...
coverage report
Name              Stmts   Miss  Cover
-------------------------------------
ool/__init__.py      49      1    98%
```

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+django >= 1.6b1, < 1.9
```
Comment (davidharrigan) [inline requirements/base.txt]: I see that you updated to 1.8 few days ago, I can make the same change here.  Looking at your Travis log, not sure why things weren't working for me >= 1.9. It's possible that I didn't have the latest version as of 5 days ago.  I'll look at this again.

**Comment by davidharrigan** [comment]
Looks like travis will need to export `SETTINGS=tests.tests.travis_settings` before running make test.  Or I can rework this PR so that `tests` is the top level.  Let me know if you would like me to re-work that.

**Comment by gavinwahl** [comment]
What is the advantage of running tests through setup.py? It seems like a lot of code.

**Comment by davidharrigan** [comment]
one advantage is `setup.py test` will resolve your dependency for you which is pretty nice - otherwise I don't think there's a real advantage.  This is something that I did for my organization, and we've been running tests through setup.py.  

I think this is too much change. I'll close this PR and open a new one that uses manage.py instead and keep the changes minimal for you to review.

**Pull request closed without merging**